components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -495,16 +501,20 @@ private static String getServiceProviderMappedUserRoles(ApplicationConfig applic
                                             allSPMappedClaims, spRequestedClaims);
 
         context.setProperty(FrameworkConstants.UNFILTERED_SP_CLAIM_VALUES, allSPMappedClaims);
-
-        if (spStandardDialect != null) {
-            setSubjectClaimForLocalClaims(tenantAwareUserName, userStore,
-                                          allLocalClaims, spStandardDialect, context);
+        if (stepConfig.isSubjectIdentifierStep()) {
```

Comment:
is this not happening in https://github.com/IsuraD/identity-framework/blob/ac7a8db89f5ebde354fc6f9b4deb3dd2228d3d9e/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandler.java#L349

Reply:
That happes from https://github.com/IsuraD/identity-framework/blob/ac7a8db89f5ebde354fc6f9b4deb3dd2228d3d9e/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthenticatedSubjectIdentifierHandler.java#L88 PostAuthenticatedSubjectIdentifierHandler. We set this subjectclaimUri value into the authentication context and read it from PostAuthenticatedSubjectIdentifierHandler. If it is not set, then we set it from the user's username.